File: 스택_큐/지현/올바른_괄호.py
def solution(s):
    stack=[]
    for i in s: 
        if not stack and i==')':
            return False
        if i=='(':
            stack.append(')')
        else:
            if stack[-1]==i:
                stack.pop()
            else:
                return False
    if stack==[]:return True
    else:return False

File: 스택_큐/[지현] 스택_큐/올바른_괄호.js
function solution(s) {
    let answer = true;
    let arr = s.split("");
    let stack = [];

    for (let i = 0; i < arr.length; i++) {
        if (arr[i] == "(") {
            stack.push(")");
        } else {
            if (stack.length == 0) {
                answer = false;
                break;
            } else {
                stack.pop();
            }
        }
    }

    if (stack.length > 0) {
        answer = false;
    }
    return answer;
}
